src/ray/util/network_util.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/util/network_util.h"
+
+#include <array>
+#include <boost/asio.hpp>
+#include <boost/asio/ip/address.hpp>
+#include <optional>
+#include <string>
+
+#include "absl/strings/str_format.h"
+
+using boost::asio::ip::tcp;
+
+namespace ray {
+
+std::string BuildAddress(const std::string &host, int port) {
+  boost::system::error_code ec;
+  auto ip_addr = boost::asio::ip::make_address(host, ec);
+  if (!ec && ip_addr.is_v6()) {
```

Comment:
I think a simpler and faster validation of finding `:` is enough: ipv4 doesn't have `:` and hostname also cannot have `:`.